### PR TITLE
Add pagination support

### PR DESCRIPTION
**Pull request by nunorafaelrocha**
This adds support for api endpoints that have pagination.

### CODE REVIEW

File: examples/Reserve.php [hunk 1]
```diff
@@ -14,6 +14,11 @@
 print_r($statistics);
 
 // Get the reserve ledger
-$ledger = $client->getReserve()->getLedger();
+$ledgerPager = $client->getReserve()->getLedger();
```
Comment (fixe) [inline examples/Reserve.php]: `$pager`?

Comment (nunorafaelrocha) [inline examples/Reserve.php]: @fixe In the same file, on line 22 we will have another `$pager` and that doesn't seems right.

Comment (fixe) [inline examples/Reserve.php]: Since this is an example you can reuse the var in L22.

File: examples/Reserve.php [hunk 2]
```diff
@@ -14,6 +14,11 @@
 print_r($statistics);
 
 // Get the reserve ledger
-$ledger = $client->getReserve()->getLedger();
+$pager = $client->getReserve()->getLedger();
 
-print_r($ledger);
```
Comment (fixe) [inline examples/Reserve.php]: Undefined. What's the output?

File: examples/Reserve.php [hunk 3]
```diff
@@ -14,6 +14,11 @@
...
+print_r($pager->getNext());
```
Comment (fixe) [inline examples/Reserve.php]: What's the output?

File: lib/Bitreserve/Paginator/Paginator.php [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+namespace Bitreserve\Paginator;
+
+/**
+ * Paginator.
+ */
+class Paginator
+{
+    /**
+     * Bitreserve Client
```
Comment (fixe) [inline lib/Bitreserve/Paginator/Paginator.php]: `.`.

File: lib/Bitreserve/Paginator/Paginator.php [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+    /**
+     * Request headers.
+     *
+     * @var array
+     */
+    protected $headers;
+
+    /**
+     * Range offset
```
Comment (fixe) [inline lib/Bitreserve/Paginator/Paginator.php]: `.`

File: lib/Bitreserve/Paginator/Paginator.php [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+     *
+     * @var int
+     */
+    protected $rangeOffset = 0;
```
Comment (fixe) [inline lib/Bitreserve/Paginator/Paginator.php]: `offset`?

File: lib/Bitreserve/Paginator/Paginator.php [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+    /**
+     * Range limit.
+     *
+     * @var int
+     */
+    protected $rangeLimit = 50;
```
Comment (fixe) [inline lib/Bitreserve/Paginator/Paginator.php]: `limit`?

File: lib/Bitreserve/Paginator/Paginator.php [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+     */
+    public function getModel()
+    {
+        return $this->model;
+    }
+
+    /**
+     * Set model.
+     *
+     * @param Paginator
```
Comment (fixe) [inline lib/Bitreserve/Paginator/Paginator.php]: `string`.

Comment (fixe) [inline lib/Bitreserve/Paginator/Paginator.php]: `@return Paginator`?

File: examples/Reserve.php (hunk 3 above)
Comment (nunorafaelrocha) [inline examples/Reserve.php]: An array with a page of ledger entries.

Comment (nunorafaelrocha) [inline examples/Reserve.php]: @fixe An array with a page of ledger entries.

File: test/Bitreserve/Tests/Paginator/PaginatorTest.php [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+
+        $pager = new Paginator($client, '/path');
+
+        $this->assertEquals($data, $pager->getNext());
+    }
+
+    /**
+     * @test
+     */
+    public function shouldReturnNextResultsWithTransactionModel()
```
Comment (fixe) [inline test/Bitreserve/Tests/Paginator/PaginatorTest.php]: Rename to `...WithGivenModel`?

File: test/Bitreserve/Tests/Model/UserTest.php [hunk 10]
```diff
@@ -496,7 +496,11 @@ public function shouldReturnTransactions()
...
         $user = new User($client, array('username' => 'foobar'));
 
-        $transactions = $user->getTransactions();
+        $pager = $user->getTransactions();
+
+        $this->assertInstanceOf('Bitreserve\Paginator\Paginator', $pager);
+
+        $transactions = $pager->getNext();
 
         $this->assertCount(count($data), $transactions);
```
Comment (fixe) [inline test/Bitreserve/Tests/Model/UserTest.php]: Missing assert that `$transactions` is an array of `Transaction` models?

File: test/Bitreserve/Tests/HttpClient/Message/ResponseTest.php [hunk 11]
```diff
@@ -25,6 +25,40 @@ public function shouldReturnResponseContent()
     /**
      * @test
      */
+    public function shouldReturnContentRangeHeaderParsed()
```
Comment (fixe) [inline test/Bitreserve/Tests/HttpClient/Message/ResponseTest.php]: Rename to `shouldReturnParsedContentRangeHeader`

File: test/Bitreserve/Tests/Model/CardTest.php [hunk 12]
```diff
@@ -175,7 +175,11 @@ public function shouldReturnTransactions()
 
         $card = new Card($client, $cardData);
 
-        $transactions = $card->getTransactions();
+        $pager = $card->getTransactions();
+
+        $this->assertInstanceOf('Bitreserve\Paginator\Paginator', $pager);
+
+        $transactions = $pager->getNext();
```
Comment (fixe) [inline test/Bitreserve/Tests/Model/CardTest.php]: Missing assert that `$transactions` is an array of `Transaction` models?

File: test/Bitreserve/Tests/Model/UserTest.php (hunk 10 above)
Comment (nunorafaelrocha) [inline test/Bitreserve/Tests/Model/UserTest.php]: Is done in L508?

Comment (nunorafaelrocha) [inline test/Bitreserve/Tests/Model/UserTest.php]: @fixe Is done in L508?

File: test/Bitreserve/Tests/Model/CardTest.php (hunk 12 above)
Comment (nunorafaelrocha) [inline test/Bitreserve/Tests/Model/CardTest.php]: @fixe Isn't done in L187?

File: lib/Bitreserve/Paginator/Paginator.php [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+    }
+
+    /**
+     * Get next results page.
+     *
+     * @return mixed
+     */
+    public function getNext()
+    {
+        $rangePositions = $this->getNextRange();
```
Comment (fixe) [inline lib/Bitreserve/Paginator/Paginator.php]: Rename to `range`.

File: lib/Bitreserve/Paginator/Paginator.php [hunk 14]
```diff
@@ -0,0 +1,221 @@
...
+        }, $data);
+    }
+
+    /**
+     * Build range header.
+     *
+     * @param int $start Range start position.
+     * @param int $end Range end position.
+     *
+     * @return [type] [description]
```
Comment (fixe) [inline lib/Bitreserve/Paginator/Paginator.php]: Missing doc.

File: test/Bitreserve/Tests/Model/CardTest.php (hunk 12 above)
Comment (fixe) [inline test/Bitreserve/Tests/Model/CardTest.php]: Right.

File: test/Bitreserve/Tests/Model/UserTest.php (hunk 10 above)
Comment (fixe) [inline test/Bitreserve/Tests/Model/UserTest.php]: Right.

**Pull request merged**